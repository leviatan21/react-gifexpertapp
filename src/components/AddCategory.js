import React, { useState } from 'react';
import PropTypes from 'prop-types'
//	rafcp
export const AddCategory = ( { setCategories } ) => {
	//
	const initialState = "";
	const [ inputValue, setInputValue ] = useState( initialState );
	//
	const handleInputChange = ( e ) => {
		//	console.log( 'handleInputChange', e.target.value );
		setInputValue( e.target.value );
	}
	//
	const handleSubmit = ( e ) => {
		e.preventDefault( );
		//	console.log( 'handleSubmit', inputValue );
		if ( inputValue.trim().length > 2 ) {
			setCategories( state => [ inputValue, ...state ] );
			setInputValue( initialState );
		}
	}
	//
	return (
		<form onSubmit={ handleSubmit }>
			<input 
				type="text"
				value={ inputValue }
				onChange={ handleInputChange }
				placeholder="..."
			/>
		</form>
	)
}
//
AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
}