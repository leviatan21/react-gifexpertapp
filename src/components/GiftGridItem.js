import React from 'react';
import PropTypes from 'prop-types'
//
export const GiftGridItem = ( { title, url } ) => {
	//
	return (
		<div className="card animate__animated animate__fadeIn animate__delay-2s">
			<img src={ url } alt={ title } />
			<p>{ title }</p>
		</div>
	)
}
//
GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}