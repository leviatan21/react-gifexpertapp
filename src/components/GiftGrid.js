import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GiftGridItem } from './GiftGridItem';
//
export const GiftGrid = ( { category } ) => {
	//
	const { data:images, loading } = useFetchGifs( category );
	//
	return (
		<>
			<h3 className="animate__animated animate__fadeIn">{ category }</h3>
			{ loading && <p className="animate__animated animate__flash">Cargando</p> }
			<div className="card-grid">
				{ images.map( (gif) => <GiftGridItem key={ gif.id } { ...gif } /> ) }
			</div>
		</>
	)
}
