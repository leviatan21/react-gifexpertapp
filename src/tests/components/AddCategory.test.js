
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
//
describe( 'Puebas en <AddCategory />', () => {
	//
	const valueNew 		= 'Goku';
	const setCategories = jest.fn( );
	let wrapper			= shallow( <AddCategory setCategories={ setCategories } /> );
	
	//
	beforeEach( () => {
		jest.clearAllMocks( );
		wrapper			= shallow( <AddCategory setCategories={ setCategories } /> );
	});
	//
	test( 'Componente AddCategory', ( ) => {
		expect( wrapper ).toMatchSnapshot();
	} );
	//
	test( 'Componente Form existe', ( ) => {
		const Form = wrapper.find( 'form' );
		expect( Form.length ).toBe( 1 );
	} );
	//
	test( 'Componente Form vacio', ( ) => {
		wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } );
		expect( setCategories ).not.toHaveBeenCalled( );
	} );
	//
	test( 'Componente Input existe', ( ) => {
		const Input = wrapper.find( 'input' );
		expect( Input.length ).toBe( 1 );
	} );
	//
	test( 'Componente Input change', ( ) => {
		wrapper.find( 'input' ).simulate( 'change', { target: { value: valueNew } } );
	} );
	//
	test( 'Componente Form submit', ( ) => {
		const input = wrapper.find( 'input' )
		input.simulate( 'change', { target: { value: valueNew } } );
		wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } );
		expect( setCategories ).toHaveBeenCalled( );
		expect( setCategories ).toHaveBeenCalledTimes( 1 );
		expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
		const value = input.prop( 'value' );
		expect( value ).toBe( "" );
	} );
} );
