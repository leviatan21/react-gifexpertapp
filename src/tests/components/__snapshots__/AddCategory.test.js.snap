// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <AddCategory /> Componente AddCategory 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="..."
    type="text"
    value=""
  />
</form>
`;
